src/modules/python/pbs_hooks/PBS_power.PY

```diff
@@ -319,40 +344,73 @@ if this_event.type == pbs.PERIODIC:
             nodes[host]["poweroff"] = 0
             nodes[host]["poweron"] = 0
 
+        rs_list = []
+        if vnode.resv:
+            resv_str = str(vnode.resv)
+            rs_list = resv_str.split(",")
+        
         if nodes[host]["can_power_off"]:
             # See if the node is actually free
             if vnode.state != pbs.ND_FREE:
                 nodes[host]["poweroff"] = 0
             else:
                 # See if there are any reservations on the vnode
                 # Reservations further in time can be ignored.
-                if vnode.resv:
-                    for resid in vnode.resv:
-                        resv = resvlist[resid]
-                        if resv.state == pbs.RESV_STATE_RUNNING or \
-                                resv.state == pbs.RESV_STATE_DELETED or \
-                                resv.state == pbs.RESV_STATE_BEING_ALTERED or \
-                                resv.state == pbs.RESV_STATE_DELETING_JOBS:
+                for resid in rs_list:
+                    resv = resvlist[resid.lstrip()]
+                    if resv.reserve_state == pbs.RESV_STATE_RUNNING or \
+                            resv.reserve_state == pbs.RESV_STATE_DELETED or \
+                            resv.reserve_state == pbs.RESV_STATE_BEING_ALTERED or \
+                            resv.reserve_state == pbs.RESV_STATE_DELETING_JOBS:
```

Comment:
Instead of this, do
rstates = [pbs.RESV_STATE_RUNNING, pbs.RESV_STATE_DELETED, pbs.RESV_STATE_BEING_ALTERED, pbs.RESV_STATE_DELETING_JOBS]
if resv.reserve_state in rstates:

Reply:
Okay.